sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2507,7 +2507,7 @@ object GpuOverrides extends Logging {
               TypeEnum.FLOAT, TypeEnum.DOUBLE, TypeEnum.DATE, TypeEnum.TIMESTAMP,
               TypeEnum.STRING, TypeEnum.DECIMAL), "Unsupported as array index."),
           TypeSig.all)),
-      (in, conf, p, r) => new BinaryExprMeta[ElementAt](in, conf, p, r) {
+      (in, conf, p, r) => new ElementAtMeta(in, conf, p, r) {
```

Comment:
Nit: It's weird to have a dedicated meta class to an expression that needs to be heavily overridden to be used properly.  Suggest moving this anonymous class override code to something like a common ElementAtMetaBase class that the specific ElementAtMeta shim classes derive from.